src/functions-and-arrays.js

```diff
@@ -1,41 +1,128 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+  if ( a >= b ) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(array) {
+  let longest;
+  if ( array.length === 0 ) { 
+    return null
+  } else {
+    for ( let i = 0; i < array.length; i++ ) {
+      const word = array[i];
+      if ( !longest || longest.length < word.length ) {
+        longest = word;
+      }
+    }
+    return longest;
+  }
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(array) {
+  let total = 0;
+  if ( array.length === 0 ) {
+    return 0;
+  } else {
+    for ( let i = 0; i < array.length; i++ ) {
+      total += array[i]
+    } 
+  }
+  return total;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(array) {
+  let total = 0;
+  if ( array.length === 0 ) {
+    return 0;
+    } else {
+      for ( let i = 0; i < array.length; i++ ) {
+        const item = array[i]
+        if ( typeof item === "string" ) {
+          total += item.length;
+        } else if ( typeof item === "number" ) {
+          total += item
+        } else if ( typeof item === "boolean" ) {
+          total += item
+        } else {
+          throw new Error ("Unsupported data type sir or ma'am");
+        }
+      }
+    }
+    return total;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(array) {
+
+  if (array.length === 0) {
+    return null;
+  };
+
+  let sum = sumNumbers(array);
+
+  return sum / array.length;
+
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(array) {
+  let total = 0;
+  if ( array.length === 0 ) {
+    return null;
+  } else {
+  array.forEach(element => {
+    total += element.length
+    });
+  }
+  let avg = total/array.length
+  return avg
+ }
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(arr) {
+  let total = 0;
+  if ( arr.length === 0 ) {
+    return null;
+    } else {
+      for ( let i = 0; i < arr.length; i++ ) {
+        const item = arr[i]
+        if ( typeof item === "string" ) {
+          total += item.length;
+        } else if ( typeof item === "number" ) {
+          total += item
+        } else if ( typeof item === "boolean" ) {
+          total += item
+        } else {
+          throw new Error ("Unsupported data type sir or ma'am");
+        }
+      }
+    }
+    let avg = total/arr.length
+    return avg
```

Comment:
```suggestion
    }
    let avg = sum(arr)/arr.length
    return avg
```

A ver, si te fijas el metodo sum es exactamente el mismo codigo, pues entonces lo que hago es ejecutarlo, repito, ejecutarlo pasandole el array que me llega! y asi lo tenemos!!!